tutorials/plot_background_statistics.py

```diff
@@ -0,0 +1,496 @@
+
+###############################################################################
+# Hypothesis testing
+# ------------------
+# Null hypothesis
+# ^^^^^^^^^^^^^^^
+# From `Wikipedia <https://en.wikipedia.org/wiki/Null_hypothesis>`_:
+#
+#     In inferential statistics, a general statement or default position that
+#     there is no relationship between two measured phenomena, or no
+#     association among groups.
+#
+# We typically want to reject a **null hypothesis** with
+# at some probability level (e.g., p < 0.05).
+# To think about what this means, let's follow the illustrative example from
+# [1]_ and construct a toy dataset consists of a 40 x 40 square with a "signal"
+# present in the center (at pixel [20, 20]) with white noise
+# added and a 5-pixel-SD normal smoothing kernel applied.
+width = 40
+n_subjects = 10
+signal_mean = 100
+signal_sd = 100
+noise_sd = 0.01
+gaussian_sd = 5
+sigma = 1e-3  # sigma for the "hat" method
+threshold = -stats.distributions.t.ppf(0.05, n_subjects - 1)
+n_permutations = 1024  # number of clustering permutations (1024 for exact)
+n_src = width * width
+
+# For each "subject", make a smoothed noisy signal with a centered peak
+rng = np.random.RandomState(42)
+X = noise_sd * rng.randn(n_subjects, width, width)
+# Add a signal at the dead center
+X[:, width // 2, width // 2] = signal_mean + rng.randn(n_subjects) * signal_sd
+# Spatially smooth with a 2D Gaussian kernel
+size = width // 2 - 1
+gaussian = np.exp(-(np.arange(-size, size + 1) ** 2 / float(gaussian_sd ** 2)))
+for si in range(X.shape[0]):
+    for ri in range(X.shape[1]):
+        X[si, ri, :] = np.convolve(X[si, ri, :], gaussian, 'same')
+    for ci in range(X.shape[2]):
+        X[si, :, ci] = np.convolve(X[si, :, ci], gaussian, 'same')
+
+###############################################################################
+# In this case, a null hypothesis we could test is for each voxel is:
+#
+#     There is no difference between the mean value and zero
+#     (:math:`H_0: \mu = 0`).
+#
+# The alternative hypothesis, then, is that the voxel has a non-zero mean.
+# This is a *two-tailed test* because this means the mean could be less than
+# or greater-than zero (whereas a *one-tailed test* would test only one of
+# these possibilities, i.e. :math:`H_0: \mu \geq 0` or
+# :math:`H_0: \mu \leq 0`).
+#
+# Parametric tests
+# ^^^^^^^^^^^^^^^^
+# Let's start with a **1-sample t-test**, which is a standard test
+# for differences in paired sample means. This test is **parametric**,
+# as it assumes that the underlying sample distribution is Gaussian, and is
+# only valid in this case. (This happens to be satisfied by our toy dataset,
+# but is not always satisfied for neuroimaging data.)
+#
+# In the context of our toy dataset, which has many voxels, applying the
+# 1-sample t test is called a *mass-univariate* approach as it treats
+# each voxel independently.
+
+# X needs to be a multi-dimensional array of shape (n_samples, n_time, n_space)
```

Comment:
Really? Normally the space (channel) dimension comes before the time dimension in MNE.